lib/Rfc1035StubResolver.php

```diff
@@ -232,8 +256,25 @@ public function query(string $name, int $type): Promise
         }
 
         $promise = call(function () use ($name, $type) {
-            if (!$this->config) {
-                yield $this->reloadConfig();
+            if (!$this->configStatus) {
+                try {
+                    yield $this->reloadConfig();
+                } catch (ConfigException $e) {
+                    $this->configStatus = self::CONFIG_FAILED;
```

Comment:
Issuing a warning here would probably be a good idea.

Reply:
I'm not a fan of logging via PHP errors, but I guess it's ok anyway, done

Reply:
Do you have a better suggestion?